### DISCUSSION TITLE
how to Tune PID for real-robot

### DISCUSSION THREAD

**Discussion by thanhtruongphan** [Ekumen-OS/andino#275] [General]
because I am using the same motor as Andino is using but different type.

the default PID parameters seem to be not suitable for my Motor, but i dont know how to Tune it?

someone help me, please.

-thank you-

**Comment by francocipollone** [comment]
> because I am using the same motor as Andino is using but different type.
> 
> the default PID parameters seem to be not suitable for my Motor, but i dont know how to Tune it?
> 
> someone help me, please.
> 
> -thank you-

What do you mean with different type? Can you provide more specs on the new motor?

If the connections are the same (encoders / forward / backward) the only thing you should change is the encoders ticks per revolution value: See https://github.com/Ekumen-OS/andino/blob/humble/andino_description/config/andino/hardware.yaml#L6

You should insert there the number of encoder ticks per revolution of your motor. You can find it in the datasheet or even better, measure yourself using some script, you can use something like https://github.com/Ekumen-OS/andino/tree/humble/andino_firmware#test-it

**Reply by thanhtruongphan** [reply]
yah, happy for your reply...

My motor is https://www.thegioiic.com/ga25-370-dong-co-giam-toc-co-encoder-12vdc-130-rpm-truc-4mm

It has 11 pulse incredemental encoder hall. A gearbox ratio 45:1, so with mode x4 quadrature encoder: 45x11x4 = 1980 ppr

> If the connections are the same (encoders / forward / backward) the only thing you should change is the encoders ticks per revolution value: See https://github.com/Ekumen-OS/andino/blob/humble/andino_description/config/andino/hardware.yaml#L6

yah, I did it.

But I have two problems with control motor. 
- motor always run with maximum velocity. (I think may be by PID, I have a little experience with turn PID. When PID is not correct, it always turn with maximum)

- when i use **```teleop```** to control **left/right** rotation, it only works about 0.5 seconds (in a flash), then it rotates forward at maximum speed

**Reply by francocipollone** [reply]
Sorry I missed this response. I wonder if you were able to solve it. @thanhtruongphan

**Reply by Russ76** [reply]
Be very careful with pins on the L298N. I had to swap my center connectors, then the speed control came up. These are in1 and in2, and swapping in3 and in4. The outside pins, En1 and En2 are easier to get connected right. It's easier to troubleshoot speed problems with the serial interface: connecting just to the Arduino, (via laptop computer), not using the Andino software on the Raspberry Pi at all for this.
